src/modules/Dropdown/Dropdown.js

```diff
@@ -631,13 +634,19 @@ export default class Dropdown extends Component {
     const options = this.getMenuOptions()
     const lastIndex = options.length - 1
 
+    // Prevent infinite loop
+    if (_.every(options, 'disabled')) return
+
     // next is after last, wrap to beginning
     // next is before first, wrap to end
     let nextIndex = selectedIndex + offset
     if (nextIndex > lastIndex) nextIndex = 0
     else if (nextIndex < 0) nextIndex = lastIndex
 
     this.setState({ selectedIndex: nextIndex })
+
+    if (options[nextIndex].disabled) return this.moveSelectionBy(offset)
```

Comment:
This will set state every time it skips a disabled item.  It would be ideal to only set state once a valid next index was found.  What if we also allow passing a `startIndex`?

```js
moveSelectionBy = (offset, startIndex = this.state.selectedIndex) => {
  // update selectedIndex usage
}
```

Then we can recurse before setting state:

```js
if (options[nextIndex].disabled) return this.moveSelectionBy(offset, nextIndex)

this.setState({ selectedIndex: nextIndex })
```